src/Features/Core/Portable/Completion/CompletionRules.cs

```diff
@@ -180,7 +188,7 @@ public TextChange GetTextChange(CompletionItem selectedItem, char? ch = null, st
 
         protected virtual bool IsCommitCharacterCore(CompletionItem completionItem, char ch, string textTypedSoFar)
         {
-            return false;
+            return s_defaultCommitCharacters.Contains(ch);
```

Comment:
I'm assuming this is a hot path right? If so, I'd consider not using LINQ's Contains to find this - this basically goes through the ICollection<T>.Contains implementation on arrays, which is an indirection due to the way arrays implement their generic interfaces (as they aren't really generic).

I'd consider Array.BinarySearch, the later of which I found was 3 times as fast as Contains (and 2 times as fast as Array.IndexOf) in brief tests that I tried, both on misses and hits.